### DISCUSSION TITLE
OpenOCD cannot find libgcc_s_dw2-1.dll

### DISCUSSION THREAD

**Discussion by Ing-Dom** [earlephilhower/arduino-pico#456] [Q&A] [answered]
when I call the openocd.exe delivered with 1.10.0 i get a error that libgcc_s_dw2-1.dll is missing.

Any idea how to solve that?

**Comment by earlephilhower** [comment]
The Windows and Linux DLL/.so requirements for openocd are quite a pain.  Looks like something busted in the (I thought) unchanging Docker image I use to do the cross-builds.

Can you confirm it is OK with 1.9.13?  If so, I can simply replace the exe with the one from that and do a minor rev.

**Reply by Ing-Dom** [reply]
I don't know, but I will get 1.9.13 and check it

1.9.13 also fails due to missing libgcc_s_dw2
1.9.1 fails due to missing libusb, when i add libusb from 1.10.0, it fails due to mssinig libgcc_s_dw2


precompiled openocd from https://www.digikey.de/de/maker/projects/raspberry-pi-pico-and-rp2040-cc-part-2-debugging-with-vs-code/470abc7efb07432b82c95f6f67f184c0 works.

But debugging does not work, breakpoints do not hit...

**Reply by earlephilhower** [reply]
Ah, poop.  The libgcc is the mingw stuff and I think I can simply include it in the  package path.  Hopefully I will have time this weekend to look at it.  I do my work under Linux and build using the MINGW tools which has done great for the actual compiler/etc., but for OpenOCD with many dependencies is not so nice...

**Reply by Ing-Dom** [reply]
if you need me to test something just tell - I will go for the digikey openocd in the meantime.

**Reply by earlephilhower** [reply]
I did some tests on a little desktop I had sitting around and can repro the failure.

After looking at what's going on, it seems to be a problem of using an updated libusb.dll with an older mingw installation.  The libusb.dll wants a different gcc.dll version than the rest of the app, and barfs.  The reason for using a manually updated libusb.dll is that the one shipped when I build the x-compile image had a bad bug and was reported as unusable by many folks under Windows.

This will require a rebuild of the Docker image to fix.  It may be a little time before I can make that magic happen...

**Reply by Ing-Dom** [reply]
thanks for investigating.

**Comment by earlephilhower** [comment, marked as answer]
Fixed in 1.12.0
